qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -109,170 +108,123 @@ class Grover:
-    ], skip=1)  # skip the argument 'self'
     def __init__(self,
-                 oracle: Union[QuantumCircuit, Statevector],
-                 good_state: Optional[Union[Callable[[str], bool],
-                                            List[str], List[int], Statevector]] = None,
-                 state_preparation: Optional[QuantumCircuit] = None,
-                 iterations: Union[int, List[int]] = 1,
+                 iterations: Union[List[int], Iterator[int], float] = 1.2,
                  sample_from_iterations: bool = False,
-                 post_processing: Callable[[List[int]], List[int]] = None,
-                 grover_operator: Optional[QuantumCircuit] = None,
                  quantum_instance: Optional[Union[QuantumInstance, Backend, BaseBackend]] = None,
-                 incremental: bool = False,
-                 num_iterations: Optional[int] = None,
-                 lam: Optional[float] = None,
-                 rotation_counts: Optional[List[int]] = None,
-                 mct_mode: Optional[str] = None,
                  ) -> None:
         # pylint: disable=line-too-long
         r"""
         Args:
-            oracle: The oracle to flip the phase of good states, :math:`\mathcal{S}_f`.
-            good_state: A callable to check if a given measurement corresponds to a good state.
-                For convenience, a list of bitstrings, a list of integer or statevector can be
-                passed instead of a function. If the input is a list of bitstrings, each bitstrings
-                in the list represents a good state. If the input is a list of integer,
-                each integer represent the index of the good state to be :math:`|1\rangle`.
-                If it is a :class:`~qiskit.quantum_info.Statevector`, it represents a superposition
-                of all good states.
-            state_preparation: The state preparation :math:`\mathcal{A}`. If None then Grover's
-                 Search by default uses uniform superposition.
             iterations: Specify the number of iterations/power of Grover's operator to be checked.
-                It the number of solutions is known, this should be an integer specifying the
-                optimal number of iterations (see ``optimal_num_iterations``). Alternatively,
-                this can be a list of powers to check.
+                * If a list, all the powers in the list are run in the specified order.
+                * If an iterator, the powers yielded by the iterator are checked, until a maximum
+                number of iterations or maximum power is reached.
+                * If a float larget than 1, this specifies the growth rate. The iterator is set to
+                yield the increasing powers of this number.
+
+                If the number of solutions is known, this should be set to a list with a single
+                entry, namely the optimal number of iterations, see
```

Comment:
We can no longer have single int then, like the default before, but it has to be wrapper as a list since float has been added

Reply:
Right, there's a tradeoff between two common use-cases: giving a fixed power, when the solution is known, and giving a growth rate `lam` (using `lam^1, lam^2, etc`) when the solution is unknown. We can do
1. accept `int` and `float`: very confusing because `2.0` is not the same as `2`
2. accept `int`: then we have
    ```
    Grover(power)  # for a fixed number, nice to do now
    Grover(map(lambda x: int(lam ** x), itertools.count(1)))  # for a growth rate, pretty tedious
    ```
3. accept `float`:
    ```
    Grover([power])  # for a fixed number, still little effort
    Grover(lam)  # also easy
    ```
    
I went with option 3 since it seemed the best compromise

Reply:
I know in AE you just changed things so that a single int did not need to be wrapped in a list, which is a bit of a pain. Aside from further complicating things by say allowing None or string 'single' and defaulting to this, so in general you would never need to do it explicitly, and its distinguishable from float. Of course this then is the old default so may not be what you want.

Reply:
It's updated to have two argument: `iterations` and `growth_rate`.